cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -394,13 +396,124 @@ func deploymentNameForEnv(envName string, clock clock.Clock) string {
 	return name[len(name)-cArmDeploymentNameLengthMax:]
 }
 
+// prevDeploymentResult looks and finds a previous deployment for the current azd project.
+func (p *BicepProvider) prevDeploymentResult(
+	ctx context.Context,
+	scope infra.Scope,
+) (*armresources.DeploymentExtended, error) {
+	deployments, err := p.findCompletedDeployments(ctx, p.env.GetEnvName(), scope, "")
+	// findCompletedDeployments returns error if no deployments are found
+	// No need to check for empty list
+	if err != nil {
+		return nil, err
+	}
+
+	if len(deployments) > 1 {
+		// If more than one deployment found, ignore the prev-deployment
+		return nil, fmt.Errorf("more than one previous deployment match.")
+	}
+
+	return deployments[0], nil
+}
+
+// applyParameters returns parameters in the same form as returned by a previous deployment, where parameters values have
+// been applied to the template's parameters.
+func applyParameters(
+	templateParameters azure.ArmTemplateParameterDefinitions, params azure.ArmParameters) map[string]interface{} {
+	result := make(map[string]interface{}, len(templateParameters))
+	for paramName, paramDefinition := range templateParameters {
+		data := make(map[string]interface{}, 2)
+		data["type"] = paramDefinition.Type
+		data["value"] = paramDefinition.DefaultValue
+		if param, exists := params[paramName]; exists {
+			data["value"] = param.Value
+		}
+		result[paramName] = data
+	}
+	return result
+}
```

Comment:
How do we ensure the ordering of the params in the map to ensure consistent comparison?

Reply:
we don't rely on order for comparison.
We only check the value for each key.  Order is not relevant as it does not make Azure RM to update things